test/runtests.jl

```diff
@@ -102,7 +102,15 @@ cd(@__DIR__) do
     #   * https://github.com/JuliaLang/julia/pull/29384
     #   * https://github.com/JuliaLang/julia/pull/40348
     n = 1
-    if net_on
+    JULIA_TEST_USE_MULTIPLE_WORKERS = get(ENV, "JULIA_TEST_USE_MULTIPLE_WORKERS", "") |>
+                                      strip |>
+                                      lowercase |>
+                                      s -> tryparse(Bool, s) |>
+                                      x -> x === true
+    # If the `JULIA_TEST_USE_MULTIPLE_WORKERS` environment variable is set to `true`, we use
+    # multiple worker processes regardless of the value of `net_on`.
+    # Otherwise, we use multiple worker processes if and only if `net_on` is true.
+    if net_on || JULIA_TEST_USE_MULTIPLE_WORKERS
         n = min(Sys.CPU_THREADS, length(tests))
         n > 1 && addprocs_with_testenv(n)
```

Comment:
Unrelated to this change, but `addprocs_with_testenv(n)` seems wrong; you get the exact count of CPU_THREADS, then *add* it to the current number of workers (1), so the end result is CPU_THREADS + 1 workers.

I noticed about this by seeing output like:

```
LinearAlgebra/givens               (25) |    18.32 |   1.32 |  7.2 |    1831.89 |   353.30
```
On a 24 logical cores machine.

It seems to be that it'd make more sense if insteadd of *adding* stuff we could specify it exactly; e.g. --procs 24 would use 24, not 25, and be consistent with that everywhere else too.

Reply:
This discussion is not really related to this PR. If you want, you can open a new issue to discuss how many worker processes `Base.runtests` should launch.